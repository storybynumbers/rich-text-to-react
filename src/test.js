import React from 'react';
import renderer from 'react-test-renderer';
import RichTextToReact from './'
//import SampleData from '../example/src/data'
import SampleRichText from '../example/src/App';

describe('RichTextToReact', () => {
  it('is truthy', () => {
    expect(RichTextToReact).toBeTruthy()
  })
})

test('Matches snapshot', () => {
  const component = renderer.create(
    <SampleRichText />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});