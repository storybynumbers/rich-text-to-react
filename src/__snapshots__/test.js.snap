// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  style={
    Object {
      "padding": 20,
    }
  }
>
  <h1>
    Rich text to React renderer
  </h1>
  <h2>
    Default rendering
  </h2>
  <p>
    Hello, world. 
    <strong>
      This is bold text
    </strong>
    . This is
    <em>
       italic
    </em>
    . This is 
    <em>
      <strong>
        bold and italic
      </strong>
    </em>
    . 
    <u>
      Underlined
    </u>
    .
  </p>
  <p>
    The start of a new paragraph 
    <a
      href="https://eyefilm.nl"
    >
      with link
    </a>
    .
  </p>
  <ul>
    <li>
      <p>
        List item one
      </p>
    </li>
    <li>
      <p>
        List item two
      </p>
    </li>
  </ul>
  <p>
    
    <span
      style={
        Object {
          "border": "1px solid #d3dce0",
          "fontFamily": "monospace",
          "margin": "0px 5px",
          "padding": "0 .25rem 0 .75rem",
        }
      }
    >
      inline: 
      entry-hyperlink
       , sys.id: 
      325gWWU4JzJ6VpFfNb7fKw
    </span>
    
  </p>
  <ol>
    <li>
      <p>
        Ordered one
      </p>
    </li>
    <li>
      <p>
        Ordered two
      </p>
    </li>
  </ol>
  <p>
    <code>
      &lt;?php
      echo 'test';
      ?&gt;
    </code>
  </p>
  <blockquote>
    <p>
      blockquoting this
    </p>
    <p>
      another paragraph with 
      <strong>
        bold
      </strong>
    </p>
  </blockquote>
  <hr />
  <p>
    
  </p>
  <h2>
    Rendering with options
  </h2>
  <p
    style={
      Object {
        "border": "#ccc 2px dotted",
        "padding": "15px",
      }
    }
  >
    Hello, world. 
    <strong
      style={
        Object {
          "color": "red",
        }
      }
    >
      This is bold text
      !
    </strong>
    . This is
    <em>
       italic
    </em>
    . This is 
    <em>
      <strong
        style={
          Object {
            "color": "red",
          }
        }
      >
        bold and italic
        !
      </strong>
    </em>
    . 
    <u>
      Underlined
    </u>
    .
  </p>
  <p
    style={
      Object {
        "border": "#ccc 2px dotted",
        "padding": "15px",
      }
    }
  >
    The start of a new paragraph 
    <a
      href="https://eyefilm.nl"
    >
      with link
    </a>
    .
  </p>
  <ul>
    <li>
      <p
        style={
          Object {
            "border": "#ccc 2px dotted",
            "padding": "15px",
          }
        }
      >
        List item one
      </p>
    </li>
    <li>
      <p
        style={
          Object {
            "border": "#ccc 2px dotted",
            "padding": "15px",
          }
        }
      >
        List item two
      </p>
    </li>
  </ul>
  <p
    style={
      Object {
        "border": "#ccc 2px dotted",
        "padding": "15px",
      }
    }
  >
    
    <span
      style={
        Object {
          "border": "1px solid #d3dce0",
          "fontFamily": "monospace",
          "margin": "0px 5px",
          "padding": "0 .25rem 0 .75rem",
        }
      }
    >
      inline: 
      entry-hyperlink
       , sys.id: 
      325gWWU4JzJ6VpFfNb7fKw
    </span>
    
  </p>
  <ol>
    <li>
      <p
        style={
          Object {
            "border": "#ccc 2px dotted",
            "padding": "15px",
          }
        }
      >
        Ordered one
      </p>
    </li>
    <li>
      <p
        style={
          Object {
            "border": "#ccc 2px dotted",
            "padding": "15px",
          }
        }
      >
        Ordered two
      </p>
    </li>
  </ol>
  <p
    style={
      Object {
        "border": "#ccc 2px dotted",
        "padding": "15px",
      }
    }
  >
    <code>
      &lt;?php
      echo 'test';
      ?&gt;
    </code>
  </p>
  <blockquote>
    <p
      style={
        Object {
          "border": "#ccc 2px dotted",
          "padding": "15px",
        }
      }
    >
      blockquoting this
    </p>
    <p
      style={
        Object {
          "border": "#ccc 2px dotted",
          "padding": "15px",
        }
      }
    >
      another paragraph with 
      <strong
        style={
          Object {
            "color": "red",
          }
        }
      >
        bold
        !
      </strong>
    </p>
  </blockquote>
  <hr />
  <p
    style={
      Object {
        "border": "#ccc 2px dotted",
        "padding": "15px",
      }
    }
  >
    
  </p>
</div>
`;
